src/main/java/com/synopsys/integration/blackduck/bdio2/Bdio2RetryAwareStreamUploader.java

```diff
@@ -29,11 +30,29 @@ public Bdio2RetryAwareStreamUploader(Bdio2StreamUploader bdio2StreamUploader) {
         this.bdio2StreamUploader = bdio2StreamUploader;
     }
 
-    public Response start(BdioFileContent header, BlackDuckRequestBuilderEditor editor)
-        throws RetriableBdioUploadException, IntegrationException {
+    public Response start(BdioFileContent header, BlackDuckRequestBuilderEditor editor, long clientStartTime, long detectTimeout)
+        throws RetriableBdioUploadException, IntegrationException, InterruptedException {
         logger.trace("Executing BDIO upload start operation; non-retryable status codes: {}", NON_RETRYABLE_EXIT_CODES);
         try {
-            return bdio2StreamUploader.start(header, editor);
+            Response response = bdio2StreamUploader.start(header, editor);
+            
+            // Retry if BlackDuck specifically told us how long to wait and we don't exceed the client timeout.
+            if (!response.isStatusCodeSuccess() && isRetryableExitCode(response.getStatusCode())) {
```

Comment:
I'm not sure...425 means the request hit the server too early so it feels like we should retry? I suppose if it's too early we won't get a retry-after header from BlackDuck but in that case we'll just return whatever response we did get and enter into the general Fibonacci based waiting which feels appropriate.

Reply:
I might be misinterpreting the description in IDETECT-4068 where it says: 

> If Detect receives one of the following HTTP responses as response to a request sent is for a Retry-After or if no response is received for HTTP response wait time, Detect should perform a backoff-retry until Detect timeout is met or an ACK is received from HUB.
> - 425 Too Early
> - 500 Internal Server Error

Does this mean we need to remove 500 from that array instead ..?

Reply:
Yes I believe so. I would say any of the codes mentioned in the issue should not be in that NON_RETRYABLE_EXIT_CODES array as it will stop us from retrying using either the new retry-after mechanism or the original Fibonacci backoff mechanism. It seems weird to me to retry on a 500 but the issue seems pretty clear on that so I'll update. Thanks for finding this.